src/pulling_gauge.rs

```diff
@@ -0,0 +1,75 @@
+use std::{collections::HashMap, fmt, sync::Arc};
+
+use crate::{
+    core::Collector,
+    proto::{Gauge, Metric, MetricFamily, MetricType},
+};
+use protobuf::RepeatedField;
+
+/// A prometheus gauge that reads the value from a provided function on every
```

Comment:
It would be better to split this docstring so that it has a short description. Like:
```rust
/// A [Gauge] that returns the value from a provided function on every collect run.
///
/// [... rest of text ...]
```

In the bottom section, you can even embed a small example. Also feel free to mention and link to the the Golang equivalent here.

Reply:
Good callout, fixed.